.github/workflows/sync-upstream.yml

```
@@ -0,0 +1,136 @@
+
+    - uses: actions/checkout@v3
+
+    - name: Add upstream
+      run: |
+        set -ex
+        PARENT=$(curl -s https://api.github.com/repos/${{github.repository}} | jq -r '.parent.clone_url')
+        git remote add upstream $PARENT
+
+    - name: Sync main
+      # This *should* be the same as git branch -f main upstream/main
+      # since we're not making any commits of our own to main
+      run: |
+        set -ex
+        # Because actions/checkout does a lazy, shallow checkout
+        # we need to use --shallow-since to make sure there's
+        # enough common history that git can tell how the two
+        # branches relate
+        #
+        # We *could* do a full checkout by setting depth: 0 above,
+        # but this is faster, especially on a large repo like this one.
+        git fetch --shallow-since="6 months" origin main
+        git checkout main
+        git pull --shallow-since="6 months" --rebase upstream main
+
+    - name: Sync ${{ github.ref_name }}
+      run: |
+        set -ex
+        git fetch --shallow-since="6 months" origin "${{ github.ref_name }}"
+        git checkout "${{ github.ref_name }}"
+        git rebase main
+
+    - name: Rebase git-annex, the feature branch
+      # This is the meatiest part of this script: rebase git-annex on top of upstream.
+      # Occasionally this step will fail -- when there's a merge conflict with upstream.
+      # In that case, you will get an email about it, and you should run these steps
+      # manually, and fix the merge conflicts that way.
+      run: |
+        set -ex
+        git fetch --shallow-since="6 months" origin git-annex
+        git checkout git-annex
+        git rebase main
+
+    - name: Construct latest version with git-annex on top
+      run: |
+        # for the latest tag vX.Y.Z, construct tag vX.Y.Z-git-annex.
+        # Only construct the *latest* release to reduce the risk of conflicts
+        # (we have to ask 'git tag' instead of the more elegant method of syncing tags
+        #  and using Github Actions' `on: push: tags: ...` because those upstream tags
+        #  *don't contain this workflow*, so there would be no way to trigger this)
+        #
+        # This will trigger release.yml to build and publish the latest version, too
+        set -e
+
+        # git fetch is supposed to get any tags corresponding to commits it downloads,
+        # but this behaviour is ignored when combined with --shallow, and there doesn't
+        # seem to be any other way to get a list of tags without downloading all of them,
+        # which effectively does --unshallow. But the GitHub API provides a shortcut, and
+        # using this saves about 30s over downloading the unshallow repo:
+        PARENT_API=$(curl -s https://api.github.com/repos/${{github.repository}} | jq -r '.parent.url')
+        PARENT_TAGS=$(curl -s "$PARENT_API"| jq -r '.tags_url')
+        RELEASE=$(curl -s "$PARENT_TAGS" | jq -r '.[0].name')
```

```suggestion
        RELEASE=$(curl -s "$PARENT_TAGS" |  jq -r 'map(.name | select(test("dev") | not)) | first')
```

Github's API *seems* (?) to run `git tag --sort=-v:refname`, which has the effect of *always* picking Gitea's `-dev` branch. e.g. at the moment the most recent is https://github.com/go-gitea/gitea/releases/tag/v1.17.0-dev, but their most recent *release* is https://github.com/go-gitea/gitea/releases/v1.16.5. Using extra jq magic we can get the most recent non-dev tag, which should be good enough.